zrml/asset-router/src/pallet_impl/named_multi_reserveable_currency.rs

```diff
@@ -91,7 +134,17 @@ impl<T: Config> NamedMultiReservableCurrency<T::AccountId> for Pallet<T> {
         value: Self::Balance,
         status: Status,
     ) -> Result<Self::Balance, DispatchError> {
-        if let Ok(currency) = T::CurrencyType::try_from(currency_id) {
+        if let Ok(asset) = T::MarketAssetType::try_from(currency_id) {
+            if T::MarketAssets::asset_exists(asset) {
+                return Err(Error::<T>::Unsupported.into());
+            }
+
+            if let Ok(currency) = T::CurrencyType::try_from(currency_id) {
+                return <T::Currencies as NamedMultiReservableCurrency<T::AccountId>>::repatriate_reserved_named(
+                    id, currency, slashed, beneficiary, value, status
+                );
+            }
+        } else if let Ok(currency) = T::CurrencyType::try_from(currency_id) {
```

Comment:
Instead of rewriting the same, why not instead of using `else if let Ok` we just use `if let Ok`?

Example:

```rust
    fn repatriate_reserved_named(
        id: &Self::ReserveIdentifier,
        currency_id: Self::CurrencyId,
        slashed: &T::AccountId,
        beneficiary: &T::AccountId,
        value: Self::Balance,
        status: Status,
    ) -> Result<Self::Balance, DispatchError> {
        if let Ok(asset) = T::MarketAssetType::try_from(currency_id) {
            if T::MarketAssets::asset_exists(asset) {
                return Err(Error::<T>::Unsupported.into());
            }
        }
        if let Ok(currency) = T::CurrencyType::try_from(currency_id) {
            return <T::Currencies as NamedMultiReservableCurrency<T::AccountId>>::repatriate_reserved_named(
                id, currency, slashed, beneficiary, value, status
            );
        }

        Err(Error::<T>::Unsupported.into())
    }
```

You might want to do the same above.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1256/commits/b5922a506c94b35d50c01a96ce803c58719281d8